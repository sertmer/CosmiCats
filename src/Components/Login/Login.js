import React, { Component } from 'react';
import './Login.scss';
import { Link } from 'react-router-dom';

export class Login extends Component {
  constructor() {
    super()

    this.state = {
      name: '',
      hasAgreed: false
    }
  }

  handleChange = (e) => {
    this.setState({ [e.target.name]: e.target.value })
  }

  handleClick = (e) => {
    this.setState({[e.target.name]: true})
  }

  render() {
    return (
      <section className='mission-card'>
        <article className='belief-card card'>
          <h2 >Our Belief</h2>
          <p>Cats are Aliens</p>
        </article>
          <form className='binding-contract card'>
            <label htmlFor='name'>Name: </label>
            <input 
              type='text'
              className='user-name' 
              name='name' 
              value={this.state.name}
              onChange={(e) => this.handleChange(e)}
            />
            <div>
              <input 
                className='agree-radio' 
                type='radio' 
                name='hasAgreed' 
                onClick={(e) => this.handleClick(e)}
              />
              <label htmlFor='hasAgreed'>I Agree</label>
            </div>
            <Link to={'/research'}>
              <button type='button'>Start Research</button>
            </Link>
          </form>
        </section>
    )
  }
}

export default Login;