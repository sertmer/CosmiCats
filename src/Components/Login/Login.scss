@import '../../_scss/variables';
@import '../../_scss/mixins';

section {
  display: flex;
  flex-direction: column;
}