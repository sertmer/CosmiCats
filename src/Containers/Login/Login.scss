@import '../../_scss/variables';
@import '../../_scss/mixins';

.mission-card {
  display: flex;
  flex-direction: column;
  width: 100%;
  justify-content: space-between;
}

.belief-header {
  font-size: 2em;
}

.card {
  @include rounded-corners(5px);
  padding: 1%;
  width:93.9vw;
  height: 60vh;
  display: flex;
  flex-direction: column;
  text-align: center;
  margin: 1%;
  background: rgb(193,193,193);
  background: linear-gradient(
    0deg, 
    rgba(193,193,193,1) 0%, 
    rgba(245,245,245,1) 100%
  );
}

.user-name-input {
  width: 60%;
  height: 10%;
  font-family: $main-font;
  font-size: 2em;
  padding: 1%;
  @include rounded-corners(5px);
  margin: 1%;
}

form {
  height: 30vh;
  display: flex; 
  flex-direction: column;
  align-items: center;
}

.login-p {
  font-size: larger;
}

.radio-div {
  display: flex;
  flex-direction: row;
  width: 40%;
  justify-content: center;
}

.agree-label {
  font-family: $main-font;
  font-size: larger;
  font-weight: bold;
  color: $alien-green;
}

.binding-contract {
  display: flex;
  flex-direction: column;
  align-items: center;
}

.complete-p {
  font-weight: bold;
  color: $alien-green;
}

.start-btn {
  width: 10em;
  @include rounded-corners(5px);
  background: rgb(69,198,141);
  background: linear-gradient(
    0deg, rgba(69,198,141,1) 0%, 
    rgba(73,210,149,1) 35%, 
    rgba(85,246,175,1) 100%
  );
  padding: 3%;
  @include rounded-corners(5px);
  font-size: 1em;
  font-family: $main-font;
  -webkit-transform: perspective(1px) translateZ(0);
  transform: perspective(1px) translateZ(0);
  -webkit-transition-duration: 0.4s;
  transition-duration: 0.4s;
  -webkit-transition-property: transform;
  transition-property: transform;
  margin: 5%;
}

.start-btn:hover {
  transform: scale(1.1);
}