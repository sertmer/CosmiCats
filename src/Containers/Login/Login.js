import React, { Component } from 'react';
import './Login.scss';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import { setUser } from '../../actions';

export class Login extends Component {
  constructor(props) {
    super(props)

    this.state = {
      name: '',
      hasAgreed: false
    }
  }

  handleChange = (e) => {
    this.setState({ [e.target.name]: e.target.value })
  }

  handleClick = (e) => {
    this.setState({[e.target.name]: true})
  }

  render() {
    const { setUser } = this.props
    const { name, hasAgreed } = this.state
    return (
      <section className='mission-card'>
        <article className='belief-card card'>
          <div className='facts'>
            <h2 className='belief-header'>Facts</h2>
            <p className='login-p'>cats were once worshipped as gifts from the gods</p>
            <p className='login-p'>science cannot explain the purr of a cat</p>
            <p className='login-p'>cats can see the unseeable</p>
            <p className='login-p'>cats always land on their feet, due to their innate anti-gravity properties</p>
            <p className='login-p'>if you hold a cats ears back, they look like aliens, dawg</p>
          </div>
          <h2 className='belief-header'>Terms of Use</h2>
          <p className='login-p'>due to the sensitive nature and the secretive acquisition of the information you are about to research, you must agree to utmost secrecy.</p>
        </article>
          <form className='binding-contract'>
            <label htmlFor='name' hidden>Name: </label>
            <input 
              type='text'
              className='user-name-input' 
              name='name' 
              value={this.state.name}
              placeholder='Your Name'
              onChange={(e) => this.handleChange(e)}
            />
            <div className='radio-div'>
              <input 
                className='agree-radio' 
                type='radio' 
                name='hasAgreed' 
                onClick={(e) => this.handleClick(e)}
              />
              <label htmlFor='hasAgreed' className='agree-label'>I Agree</label>
            </div>
            { name && hasAgreed ? 
              <Link to={'/research'}>
                <button className='start-btn' type='button' onClick={() => setUser(name)}>Start Research</button>
              </Link> 
              :
              <p className='complete-p'>Complete the form to proceed</p>
            }
          </form>
        </section>
    )
  }
}

export const mapDispatchToProps = dispatch => ({
  setUser: user => dispatch(setUser(user))
})

export default connect(null, mapDispatchToProps)(Login);