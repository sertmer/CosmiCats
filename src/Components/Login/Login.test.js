import React from 'react';
import { shallow } from 'enzyme';
import { Login } from './Login';

describe('Login', () => {

  let wrapper

  beforeEach(() => {
    wrapper = shallow(<Login />)
  })

  describe('Login unit tests', () => {
    it('should match the snapshot', () => {
      expect(wrapper).toMatchSnapshot()
    })
  })
})