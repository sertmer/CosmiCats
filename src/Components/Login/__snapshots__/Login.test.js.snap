// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Unit Tests should match the snapshot 1`] = `
<section
  className="mission-card"
>
  <article
    className="belief-card card"
  >
    <h2>
      Our Belief
    </h2>
    <p>
      Cats are Aliens
    </p>
  </article>
  <form
    className="binding-contract card"
  >
    <label
      htmlFor="name"
    >
      Name: 
    </label>
    <input
      className="user-name"
      name="name"
      onChange={[Function]}
      type="text"
      value=""
    />
    <div>
      <input
        className="agree-radio"
        name="agree"
        type="radio"
      />
      <label
        htmlFor="agree"
      >
        I Agree
      </label>
    </div>
    <Link
      to="/research"
    >
      <button
        type="button"
      >
        Start Research
      </button>
    </Link>
  </form>
</section>
`;
